File: cardprovider/src/main/java/au/com/codecandy/magma/cardprovider/model/Transaction.java
package au.com.codecandy.magma.cardprovider.model;

import java.util.Currency;
import java.util.Date;
import java.util.HashMap;
import java.util.Map;

public class Transaction {
    private Card card;
    private Double amount;
    private Currency currency;
    private String location;
    private int date;
    private String receiver;

    /** Getters **/
    public Card getCard() {
        return card;
    }

    public Double getAmount() {
        return amount;
    }

    public Currency getCurrency() {
        return currency;
    }

    public String getLocation() {
        return location;
    }

    public int getDate() {
        return date;
    }

    public String getReceiver() {
        return receiver;
    }

    /** Setters **/
    public void setCard(Card card) {
        this.card = card;
    }

    public void setAmount(Double amount) {
        this.amount = amount;
    }

    public void setCurrency(Currency currency) {
        this.currency = currency;
    }

    public void setLocation(String location) {
        this.location = location;
    }

    public void setDate() {
        this.date = new Date().getDate();
    }

    public void setReceiver(String receiver) {
        this.receiver = receiver;
    }

    // Returns the transaction as a response object
    public Map<String, Object> toObject() {
        HashMap<String, Object> map = new HashMap<>();
        map.put("card", card.toObject());
        map.put("amount", amount);
        map.put("currency", currency);
        map.put("location", location);
        map.put("date", date);
        map.put("receiver", receiver);

        return map;
    }
}

File: magma/magma/Models/Transaction/Transaction.swift
//
//  Transaction.swift
//  magma
//
//  Created by Justin Wilkin on 22/5/19.
//  Copyright © 2019 Code Candy. All rights reserved.
//

class Transaction {
    // MARK: - Properties
    private var card: Card
    private var amount: Double
    private var currency: String
    private var location: String
    private var date: String
    private var receiver: String
    
    // MARK: -
    init (card: Card, amount: Double, currency: String, location: String, date: String, receiver: String) {
        self.card = card
        self.amount = amount
        self.currency = currency
        self.location = location
        self.date = date
        self.receiver = receiver
    }
    
    // MARK: - Getters
    public func getCard() -> Card {
        return card
    }
    
    public func getAmount() -> Double {
        return amount
    }
    
    public func getCurrency() -> String {
        return currency
    }
    
    public func getLocation() -> String {
        return location
    }
    
    public func getDate() -> String {
        return date
    }
    
    public func getReceiver() -> String {
        return receiver
    }
}
